tslint.json

```diff
@@ -130,9 +130,11 @@
     "no-host-metadata-property": true,
     "no-input-rename": true,
     "no-output-rename": true,
+    "no-unused-variable": [true, {"ignore-pattern": "^_"}],
```

Comment:
Not sure if adding leading underscore on unused variables is appropriate. Feels like replacing with _ is sufficient

Reply:
@dingyuchen 
I used the regex `^_` because I feel that sometimes named unused variable may be quite useful.
For example the `originalString` and `header` here:
```typescript
const matches = this.sectionRegex.exec(unprocessedContent);
if (matches) {
  const [_originalString, _header, description] = matches;
  ...
}
```
I feel that leaving them as `_originalString, _header` instead of `_, _` will improve readability.
What do you think about this?

Reply:
I think that for the purpose of documentation, the output should be annotated via comments. eg wrap `this.sectionRegex.exec` in a function and specify the output via jsdocs comments.
Tbh I'm ok with both. A leading underscore just seems like a very arbitrary and perhaps unnecessary convention.
Can ignore if the rest are fine with it @seanlowjk @kkangs0226

Reply:
I'm ok with keeping the leading underscore, but @dingyuchen 's idea sounds good as well. I think there might be more unused variables, perhaps we can keep the leading underscore first and discuss this in another PR? What do you guys think?